### PR TITLE
[RF] Avoid using transient `std::unique_ptr`

### PR DESCRIPTION
**Pull request by guitargeek**
Harmless change to circumvent errors like these when building RooFit standalone:

```
Error in <TProtoClass::FindDataMember>: data member with index 0 is not found in class tuple<TreeReadBuffer*,default_delete<TreeReadBuffer> >
Error in <CreateRealData>: Cannot find data member # 0 of class tuple<TreeReadBuffer*,default_delete<TreeReadBuffer> > for parent RooAddPdf!
Error in <TProtoClass::FindDataMember>: data member with index 1 is not found in class tuple<TreeReadBuffer*,default_delete<TreeReadBuffer> >
Error in <CreateRealData>: Cannot find data member # 1 of class tuple<TreeReadBuffer*,default_delete<TreeReadBuffer> > for parent RooAddPdf!
Error in <TProtoClass::FindDataMember>: data member with index 0 is not found in class tuple<TreeReadBuffer*,default_delete<TreeReadBuffer> >
Error in <CreateRealData>: Cannot find data member # 0 of class tuple<TreeReadBuffer*,default_delete<TreeReadBuffer> > for parent RooPolynomial!
```

### CODE REVIEW

**Comment by vepadulano** [comment]
Wait, this should have been fixed by https://github.com/root-project/root/pull/13900. If that's not the case, we need to understand why. FYI @pcanal .

**Comment by guitargeek** [comment]
From a practical point of view for RooFit, the fact that the problem is not understood should not block this PR to enter the next patch release, unless the underlying problem is fixed in the meantime.

I'll try to create an easy reproducer!

**Comment by vepadulano** [comment]
The original issue (which was "fixed" by also removing the transient unique_ptr because it was blocking the transition to C++17) is at https://github.com/root-project/root/issues/13361

I couldn't create a standalone reproducer back then, also we should probably reintroduce the transient unique_ptr in the `RBrowserData` class and see what happens.

All of this to say, I agree with you, it's outside of scope for this PR

**Comment by pcanal** [comment]
> Harmless change to circumvent errors like these when building RooFit standalone:

Can you remind me how to reproduce this (as you saw it, no need to reduce)?

**Comment by guitargeek** [comment]
You need to compile this code with the 3rd commit in the history reverted and run the tests (just like the CI does):
https://github.com/guitargeek/roofit/commits/main/

But you need a ROOT build without RooFit for that

**Comment by pcanal** [comment]
Thanks.  I am able reproduce that error.  I am investigating.

**Comment by pcanal** [comment]
This PR is a 'downgrade' is code simplicity that should not be necessary.  However the underlying fix is not ready (and won't until August), so we could temporarily merge this if need be.

**Comment by guitargeek** [comment]
Ok, thanks for the comment! Let's see when the next patch release will be scheduled

**Comment by guitargeek** [comment]
The 6.32.04 release is planned in about 10 days, according to @dpiparo.

@pcanal, will there be a fix for the underlying issue by then, or should be just merge this PR?

**Comment by pcanal** [comment]
Checking ...

**Comment by pcanal** [comment]
@guitargeek Potential good news.  I am currently unable to reproduce the problem with the master nor with the tip of v6-32-00-patches.  Can you please verify?  Thanks.

**Comment by pcanal** [comment]
@guitargeek Nevermind.  I just realized that I can still see the problem when running some of the test :(

**Comment by pcanal** [comment]
https://github.com/root-project/root/pull/16202 should fix this problem.  Can you please check?

**Comment by guitargeek** [comment]
I see the CI in that PR is still red. Should I wait for the PR to be updated such that it's green before I test? Or does it make sense to test already now?

Thanks a lot already for fixing it :+1:

**Comment by pcanal** [comment]
It makes sense to test it now to know at the very least if my independent tests are sufficient or not.

**Comment by guitargeek** [comment]
Perfect! It works!

**Comment by guitargeek** [comment]
Superseded by a PR that fixes the underlying problem:
  * https://github.com/root-project/root/pull/16202

**Comment by guitargeek** [comment]
Re-opening, because this should be merged in case #16202 doesn't get ready for the 6.34 release.

**Review by dpiparo — APPROVED** [review]

**Pull request merged**
